src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -725,4 +775,328 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
+		}
+		return undefined;
+	}
+
+	public getColumnIndex(name: string): number {
+		return this.columnNameToIndex[name];
+	}
+
+	private getFormatter = (column: any): any => {
+		return (row, cell, value, columnDef, dataContext) => {
+			let columnId = cell > 0 && this.dataSet.columnDefinitions.length > cell - 1 ? this.dataSet.columnDefinitions[cell - 1].id : undefined;
+			if (columnId) {
+				let overrideValue = this.overrideCellFn && this.overrideCellFn(row, columnId, value, dataContext);
+				let valueToDisplay = (value + '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
+				let cellClasses = 'grid-cell-value-container';
+				/* tslint:disable:no-null-keyword */
+				let valueMissing = value === undefined || value === null;
+				/* tslint:disable:no-null-keyword */
+				let isOverridden = (overrideValue !== undefined && overrideValue !== null) || (overrideValue !== 'NULL' && row !== this.tables[0].grid.getDataLength() - 1);
+				console.log(isOverridden);
+				if (valueMissing && !isOverridden) {
+					cellClasses += ' missing-value';
+				}
+				return '<span title="' + valueToDisplay + '" class="' + cellClasses + '">' + valueToDisplay + '</span>';
+			}
+			return undefined;
+		};
+	}
+
+	handleChanges(changes: { [propName: string]: any }): void {
+		let columnDefinitionChanges = changes['columnDefinitions'];
+		let activeCell = this.tables[0] ? this.tables[0].grid.getActiveCell() : undefined;
+		let hasGridStructureChanges = false;
+		let wasEditing = this.tables[0] ? !!this.tables[0].grid.getCellEditor() : false;
+
+		if (columnDefinitionChanges && !equals(columnDefinitionChanges.previousValue, columnDefinitionChanges.currentValue)) {
+			if (!this.tables[0]) {
+				this.createNewTable();
+			} else {
+				this.tables[0].grid.resetActiveCell();
+				this.tables[0].grid.setColumns(this.dataSet.columnDefinitions);
+			}
+			hasGridStructureChanges = true;
+
+			if (!columnDefinitionChanges.currentValue || columnDefinitionChanges.currentValue.length === 0) {
+				activeCell = undefined;
+			}
+			if (activeCell) {
+				let columnThatContainedActiveCell = columnDefinitionChanges.previousValue[Math.max(activeCell.cell - 1, 0)];
+				let newActiveColumnIndex = columnThatContainedActiveCell
+					? columnDefinitionChanges.currentValue.findIndex(c => c.id === columnThatContainedActiveCell.id)
+					: -1;
+				activeCell.cell = newActiveColumnIndex !== -1 ? newActiveColumnIndex + 1 : 0;
+			}
+		}
+
+		if (changes['dataRows']
+			|| (changes['highlightedCells'] && !equals(changes['highlightedCells'].currentValue, changes['highlightedCells'].previousValue))
+			|| (changes['blurredColumns'] && !equals(changes['blurredColumns'].currentValue, changes['blurredColumns'].previousValue))
+			|| (changes['columnsLoading'] && !equals(changes['columnsLoading'].currentValue, changes['columnsLoading'].previousValue))) {
+			this.setCallbackOnDataRowsChanged();
+			this.tables[0].grid.updateRowCount();
```

Comment:
this sequence of calls looks very similar to the rerender function elsewhere in this change.  should we just invoke that method from here?